clang/runtime/dpct-rt/include/dpl_extras/dpcpp_extensions.h.inc

```diff
@@ -654,7 +658,11 @@ namespace detail {
 // DPCT_DEPENDENCY_EMPTY
 // DPCT_CODE
 template <typename... _Args> constexpr auto __joint_reduce(_Args... __args) {
+#ifdef SYCL_EXT_ONEAPI_USER_DEFINED_REDUCTIONS 
+  return sycl::ext::oneapi::experimental::joint_reduce(__args...);
```

Comment:
Does sycl::ext::oneapi::experimental::joint_reduce compatible with sycl::joint_reduce?
If both user defined op and cub op are used, then all func call will go through first code path.

Reply:
Yes, `sycl::ext::oneapi::experimental::joint_reduce` compatible with `sycl::joint_reduce`, as far as I know,   if the binary operator is sycl native operator, `sycl::ext::oneapi::experimental::joint_reduce` will call `sycl::joint_reduce`